contracts/epochs/FlowEpoch.cdc

```diff
@@ -420,13 +448,13 @@ pub contract FlowEpoch {
                     }
                 case EpochPhase.EPOCHSETUP:
                     if FlowClusterQC.votingCompleted() && (FlowDKG.dkgCompleted() != nil) {
+                        self.calculateAndSetRewards()
```

Comment:
`calculateAndSetRewards` takes up the majority of the execution time because it currently has to iterate through every delegator for all the staked node operators, even if they aren't staked, because it still has to check if their stake is non-zero. So it will move it to a different place, but it at least separates it into a different transaction from the other end epoch operations to make it not quite as large since it used to be a single transaction. And then we can ideally work on a more fundamental solution in the meantime.

I'm not sure if there is a clean way to improve it fundamentally though because currently, the delegator information is stored in the `NodeRecord` resource and we can add new fields to it to track only active delegators for example. So we'd have to either do some sort of migration to delete any delegator IDs that currently don't have any stake or we'd have to make another stored dictionary separate from `NodeRecord` that tracks all delegators,  which feels kind of dirty. That is probably out of this scope of this upgrade though.

Reply:
Agreed it's out of scope for this PR. I was just curious.